### PR TITLE
streamline onboarding experience

### PR DESCRIPTION
**Pull request by Jennievon**
# :books: Linear Ticket :books:

[#LIQ-768](https://linear.app/liquality/issue/LIQ-768/streamline-onboarding-experience)

### CODE REVIEW

**Review by monokh — APPROVED** [review]

File: src/components/AnalyticsOptInModal.vue [hunk 1]
```diff
@@ -23,14 +59,14 @@
...
-
 export default {
   components: {
     Modal
   },
   data: function () {
     return {
-      loading: false
+      loading: false,
+      analyticsAccepted: null
```
Comment (bradleySuira) [inline src/components/AnalyticsOptInModal.vue:69]: In this case you can access to `state.analytics?.acceptedDate` and use `mappState` to access it as a computed property

File: src/components/AnalyticsOptInModal.vue [hunk 2]
```diff
@@ -46,19 +82,20 @@ export default {
     ...mapActions(['setAnalyticsResponse', 'initializeAnalytics']),
     close() {
       this.setAnalyticsOptInModalOpen({ open: false })
+      this.$emit('goToSetup')
     },
     async accept() {
       this.loading = true
+      this.analyticsAccepted = true
```
Comment (bradleySuira) [inline src/components/AnalyticsOptInModal.vue:89]: Not need it if you use the acceptedDate from the state

**Review by bradleySuira — CHANGES_REQUESTED** [review]

**Comment by skeremidchiev** [comment]
What is the correct behaviour of the close (x) button?

Screenshot 2022-04-26 at 13 39 25

It seems like it brings you to next step.

**Comment by Jennievon** [comment]
> What is the correct behaviour of the close (x) button?
> 
> Screenshot 2022-04-26 at 13 39 25
> 
> It seems like it brings you to next step.

Fixed now.

File: src/components/AnalyticsOptInModal.vue (hunk 2 above)
Comment (Jennievon) [reply src/components/AnalyticsOptInModal.vue]: > Not need it if you use the acceptedDate from the state

Fixed but on `reject`, the state value is `null`. Have to add a data property for accepting. Pushed a fix. Kindly review.

**Comment by Jennievon** [comment]
@skeremidchiev @bradleySuira, please review

**Pull request merged**